testsuite/tests/typing-jkind-bounds/subsumption/fuel.ml

```diff
@@ -0,0 +1,119 @@
+(* TEST
+    flags = "-extension layouts_alpha";
+    expect;
+*)
+
+type 'a my_list = 'a list = [] | ( :: ) of 'a * 'a my_list
+
+(* At the time this test is written, "fuel" is used to deal with recursive types. Two
+   types that are equal can get different jkinds due to fuel running out at different
+   places. The below types are chosen so that they are equal, but fuel runs out in
+   different places. If fuel is changed, they should be modified so that this continues
+   to be true, and all instances of them in this file should also be updated. *)
+type t : immutable_data = int list my_list list my_list list
+[%%expect {|
+type 'a my_list = 'a list = [] | (::) of 'a * 'a my_list
+type t = int list my_list list my_list list
+|}]
+type t : immutable_data = int list list list list list list
+[%%expect {|
+Line 1, characters 0-59:
+1 | type t : immutable_data = int list list list list list list
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "int list list list list list list" is immutable_data
```

Comment:
As elsewhere, I'm surprised this doesn't print out more information about the kinds

Reply:
I don't think I follow what information you expected to see here. Although I do think the error message is bad due to the `because of the definition of t at line 1`. I think it should instead say something about the kind annotation on `t`.